File: 435-non-overlapping-intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        """
        [1,2] [1,3] [2,3], [3, 4]
        
        0 -> 1
        1 -> 0, 2
        2 -> 1
        3 -> 
        """
        if len(intervals) <= 1:
            return 0

        intervals.sort()
        count = 0

        i = 0
        for j in range(1, len(intervals)):
            if intervals[i][1] > intervals[j][0]:
                # overlap
                if intervals[i][1] > intervals[j][1]:
                    # remove the ith intervals since it has a later end time
                    i = j
                    count += 1
                else:
                    count += 1
            else:
                i = j
        
        return count

File: 435-non-overlapping-intervals/non-overlapping-intervals.go
import (
    "sort"
)

func eraseOverlapIntervals(intervals [][]int) int {
    sort.Slice(intervals, func(i, j int) bool{
        return intervals[i][0] < intervals[j][0]
    })

    i := 0
    count := 0
    for j := 1; j < len(intervals); j++{
        if intervals[i][1] > intervals[j][0]{
            if intervals[i][1] > intervals[j][1]{
                i = j
            }
            count++
        } else {
            i = j
        }
    }

    return count
}